File: 2219-maximum-number-of-words-found-in-sentences/maximum-number-of-words-found-in-sentences.java
class Solution {
    public int mostWordsFound(String[] sentences) {
        int maxWords = 0;
        
        for (String sentence : sentences) {
            // Split by spaces and count words
            int wordCount = sentence.split(" ").length;
            // Update max if current is greater
            maxWords = Math.max(maxWords, wordCount);
        }
        
        return maxWords;
    }
}

File: 2219-maximum-number-of-words-found-in-sentences/maximum-number-of-words-found-in-sentences.cpp
class Solution {
public:
    int mostWordsFound(vector<string>& sentences) {
        int ma = 0;
        for (auto sentence : sentences) {
            int words = 1; // at least one word
            for (char c : sentence) {
                if (c == ' ') words++;
            }
            ma = max(ma, words);
        }
        return ma;
    }
};
